kafka-message-signing/src/main/java/com/alliander/osgp/kafka/message/signing/MessageSigner.java

```diff
@@ -104,6 +108,24 @@ public void sign(final SignableMessageWrapper<?> message) {
     }
   }
 
+  /**
+   * Signs the provided {@code producerRecord} in the header, overwriting an existing signature, if a non-null value is
+   * already set.
+   *
+   * @param message the message
+   * @param producerRecord the record to be signed
+   * @throws IllegalStateException if this message signer has a public key for signature
+   *     verification, but does not have the private key needed for signing messages.
+   * @throws UncheckedIOException if determining the bytes for the message throws an IOException.
+   * @throws UncheckedSecurityException if the signing process throws a SignatureException.
+   */
+  public <T> void signRecordHeader(final SignableMessageWrapper<T> message, final ProducerRecord<String, T> producerRecord) {
```

Comment:
Als we de record header checken hoeft het niet perse gewrapt te worden dus mag ook gewoon T zijn in plaats van SignableMessageWrapper<T>